Makefile

```diff
@@ -427,11 +426,11 @@ endif
 # For quick compile test, used smaller subset
 ALLX_DEP= $(ALL_DEP)
 
-build/src/%.o: src/%.cc
+build/src/%.o: src/%.cc | mkldnn
 	@mkdir -p $(@D)
 	$(CXX) -std=c++11 -c $(CFLAGS) -MMD -c $< -o $@
 
-build/src/%_gpu.o: src/%.cu
+build/src/%_gpu.o: src/%.cu | mkldnn
```

Comment:
does the GPU code depend on MKLDNN?

Reply:
GPU code need ndarray.h which depends on mkldnn (need mkldnn.hpp) when MKLDNN is enabled. The mkldnn target will be empty if MKLDNN is not enabled.